Improve XML writer

tnjd opened the pull request.

TheBeggs committed:
Add WIP: Node skeletons

TheBeggs committed:
Fix formatting

TheBeggs committed:
Change abstract Node to concrete with NodeType

To prevent duplication, Node is no longer abstract. Instead of
each sub node type extending Node, they are a member. This can
happen now as metadata will be stored on the NodeType, which
is specific to that type, instead of being a map of String to
String.

TheBeggs committed:
Remove redundant import

TheBeggs committed:
Add WIP: JSimXMLWriter skeleton

TheBeggs committed:
Convert writeModel to graphly

TheBeggs committed:
Convert writeClasses to graphly

TheBeggs committed:
Convert writeStations to graphly

TheBeggs committed:
Convert XMLParameter & WIP writeSourceSection to graphly

tnjd committed:
Create Distribution

TheBeggs committed:
Convert getDistributionParameter & supporting functions to graphly

TheBeggs committed:
Merge remote-tracking branch 'origin/jsim-xml-writer' into jsim-xml-writer

# Conflicts:
#	src/main/scala/model/sim/UserClass.scala

tnjd committed:
remove JSimXMLWriter

TheBeggs committed:
Add userClasses to XMLSimRepresentation

TheBeggs committed:
Fix userClassFromXML missing params

TheBeggs committed:
Revert JSIMG changes

TheBeggs committed:
Fix type in XML

Commit:
Widget windows, minimise, close

TheBeggs committed:
WIP: converting away from w3

TheBeggs committed:
Change NodeTypes & TypeSections to be JSIM spec

TheBeggs committed:
Add GUI constants to JSIM XML constants

TheBeggs committed:
Add remaining JSIM fields to Sim

TheBeggs committed:
Add remaining JSIM fields to UserClass

TheBeggs committed:
Add UnimplementedDistribution to Distribution

TheBeggs committed:
WIP: Add parseClasses & parseXML

TheBeggs committed:
Add sections to NodeTypes

TheBeggs committed:
Add makeNodeType() & make*Section() functions

tnjd committed:
Fix xml cyclic reference error

tnjd committed:
Take filename argument in represent()

tnjd committed:
Fix unbounded recursion caused by previous commit

tnjd committed:
Generate node xml from NodeType sections

TheBeggs committed:
Add default UnimplementedSection() to constructors

TheBeggs committed:
Refactor XML Reader to JSIM style

tnjd committed:
use Node instead of Elem

TheBeggs committed:
Remove redundant Configuration field

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

# Conflicts:
#	src/main/scala/io/XMLSimRepresentation.scala

TheBeggs committed:
Fix type in sections

TheBeggs committed:
Change toolbarStatusMnemonic to show nodeType

TheBeggs committed:
Add TODO

tnjd committed:
Merge branch 'main' into improve-xml-writer

TheBeggs committed:
Add parseMeasures()

tnjd committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

TheBeggs committed:
Fix style

TheBeggs committed:
Replace strings with constants

TheBeggs committed:
Simplify Color reading

tnjd committed:
Improve XML output generation

tnjd committed:
User class XML

TheBeggs committed:
Patch None bugs

TheBeggs committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

tnjd committed:
Merge remote-tracking branch 'origin/main' into improve-xml-writer

tnjd committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

tnjd committed:
Handle unimplemented node

tnjd committed:
Merge remote-tracking branch 'origin/main' into improve-xml-writer

tnjd committed:
Improve connection xml generation

mrRachar reviewed (changes requested):
I'm not sure if it's finished, Beggs asked me to take a look though

TheBeggs committed:
Implement new Sections and relevant parsing

TheBeggs committed:
Merge branch 'main' into improve-xml-writer

# Conflicts:
#	src/main/scala/ui/canvas/GraphCanvasController.scala
#	src/main/scala/ui/toolbar/VerticalToolbar.scala

TheBeggs committed:
Fix when importing a userClass with no refClass

TheBeggs committed:
Remove redundant comments

mrRachar reviewed (changes requested):
These are the final changes, discussed with @TheBeggs. @expectocode may need to implement the relevant ones to him

Ivaneres committed:
Merge branch 'main' of https://github.com/graphly/graphly into node-metadata-editor

tnjd committed:
TypeSection fixes

Ivaneres committed:
Reworked metadata panel to be permanent, placeholder widget system

TheBeggs committed:
Refactor Sim to have a Configuration field

TheBeggs committed:
Fix bug when no user classes are present

TheBeggs committed:
Complete TODO's

TheBeggs committed:
Add defaults for sections

TheBeggs committed:
Fix style

TheBeggs committed:
Fix type bugs

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

TheBeggs committed:
Fix style

Ivaneres committed:
Fixed minimise, added stylesheet for widgets, style fixes

Ivaneres committed:
Added provisional integer and float fields to widgets

Ivaneres committed:
Added integer and double fields to widgets

mrRachar reviewed (commented):
Still some hangover from @expectocode otherwise no holdup problems

tnjd committed:
Clean up comments

tnjd committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

tnjd committed:
Remove outdated comments

Ivaneres committed:
Cleaned up code, added listener for value changes

TheBeggs committed:
Remove redundant import

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Added listener for dropdown changes

TheBeggs committed:
Merge branch 'node-metadata-editor' into improve-xml-writer

# Conflicts:
#	src/main/scala/ui/canvas/FloatingPropertiesPanel.scala
#	src/main/scala/ui/canvas/PropertiesPanel.scala

TheBeggs committed:
Remove unused code

TheBeggs committed:
Add Todo for new graphics

TheBeggs committed:
Fix writing null to file on failed save

In the case we failed to convert the model to an xml representation,
we used to write the null to the file. This is particularly bad in
the event an file is overwritten, as all progress will be lost and
failure was silent. While failure is no longer fail dangerous, the
user should still be notified that the saving failed - perhaps by a
simple popup.

TheBeggs committed:
Extend QueueSection to have size & DropStrategy

TheBeggs committed:
Add optional fields to Measures

TheBeggs committed:
Make control flow more explicit

TheBeggs committed:
Use find() to get xml results

TheBeggs committed:
Remove debug prints

TheBeggs committed:
Add blocking regions to sim

TheBeggs committed:
Refactor represent()

TheBeggs committed:
Add representMeasure() & representBlockingRegion()

TheBeggs committed:
Fix RouterSection reading

TheBeggs committed:
Add classes to representNode()

TheBeggs committed:
Fix representTypeSection()

TheBeggs committed:
Add blockingRegions & measures to writer

TheBeggs committed:
Add traces to xml writer

TheBeggs committed:
Add results to xml writer

Ivaneres committed:
Implemented editing of metadata for metadata editor widgets

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

TheBeggs committed:
Fix trace encoding in xml

Ivaneres committed:
Fixed bug with xml sim settings parsing and loading sim config into side menu

TheBeggs committed:
Add missing attributes to sim writer

TheBeggs committed:
Add unimplemented node to representNode()

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Added per-node widget editor, enter to save value of textfield, refactored projection to take Sim

Ivaneres committed:
Merge branch 'improve-xml-writer' of https://github.com/graphly/graphly into improve-xml-writer

TheBeggs committed:
Fix seed being non integer value

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Basher4 committed:
Cache image objects when rendering frame

Ivaneres committed:
Added "new" option to file menu (mostly for dev, remove if necessary)

Ivaneres committed:
Merge branch 'improve-xml-writer' of https://github.com/graphly/graphly into improve-xml-writer

TheBeggs committed:
Change xml to write node rotated state

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Added unsaved changes alert to "new" and close application

Ivaneres committed:
Merge branch 'improve-xml-writer' of https://github.com/graphly/graphly into improve-xml-writer

Ivaneres committed:
Improved unsaved changes checking to include nodes

Ivaneres committed:
Added width and height redraw listeners to GraphingCanvas

Ivaneres committed:
Added unsaved changes check to "Open"

Ivaneres committed:
Added queue-specific settings

TheBeggs committed:
Remove debug prints

Ivaneres committed:
Refactored unsaved change handling to GraphCanvasController, fixed unsaved change message on after save

Ivaneres committed:
Fixed warnings

Ivaneres committed:
Changed titlebar click zone to be the entire thing

Ivaneres committed:
Removed unused import

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

# Conflicts:
#	src/main/scala/ui/canvas/GraphCanvasController.scala

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Merged in main

Ivaneres committed:
Added longField and set seed as longField

Ivaneres committed:
Offsetting pasted nodes to make them easier to see

Ivaneres committed:
Implemented save and save as

Ivaneres committed:
Added showError to GraphCanvasController and temporary error to failed saveFile

TheBeggs committed:
Add RoutingStrategies

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Added queue widgets for more nodetypes, error on bad file open

TheBeggs committed:
Fix default logging path on windows

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Commented out parametric anaylsis enabled

Ivaneres committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

TheBeggs committed:
Add helper methods to Counters

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Merged in performance

Ivaneres committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

Ivaneres committed:
Merged

Ivaneres committed:
Added "Select All" to top menu and ctrl+a shortcut

TheBeggs committed:
Change paste binding to be more intuitive

TheBeggs committed:
Fix bug where QueueSection's wouldn't paste

TheBeggs committed:
Fix Counter not syncing with model

Closes #33

TheBeggs committed:
Fix bug where pasted nodes names would collide

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Basher4 committed:
Update icons and make it artifucktable

Basher4 committed:
Fix node color generation

Basher4 committed:
Remove useless prints

Basher4 committed:
Set traces to be 30% opque

Ivaneres committed:
Added better unsaved changes message on exit

Ivaneres committed:
Added tooltips to metadata editor

Ivaneres committed:
Switch javafx tooltip to scalafx in widgets, reduced time to 500ms in ToolbarButton

Ivaneres committed:
Moved widgetPanel package out of canvas

TheBeggs committed:
Fix paste name collision

TheBeggs committed:
Add Fork & Join sections

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

Ivaneres committed:
Added routing strategies support

Ivaneres committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

Ivaneres committed:
Added forkWidget

Ivaneres committed:
Temporary new node implementation

tnjd committed:
Remove old comments

tnjd committed:
Merge branch 'improve-xml-writer' of github.com:graphly/graphly into improve-xml-writer

TheBeggs committed:
Add defaults for all nodeSections

TheBeggs committed:
Remove Terminal button

TheBeggs committed:
Add widgets for all compatible sections

TheBeggs committed:
Remove debug prints

TheBeggs committed:
Merge remote-tracking branch 'origin/improve-xml-writer' into improve-xml-writer

# Conflicts:
#	src/main/scala/ui/AppMainSceneView.scala

TheBeggs committed:
Fix nodeTypes references

NodeTypes were referencing the same
object in certain edge cases. Lazy
evaluation is now used instead.
